src/main/kotlin/ar/edu/unahur/obj2/vendedores/Vendedor.kt

```diff
@@ -3,49 +3,67 @@ package ar.edu.unahur.obj2.vendedores
-  fun esVersatil() =
-    certificaciones.size >= 3
-      && this.certificacionesDeProducto() >= 1
-      && this.otrasCertificaciones() >= 1
+    // En las funciones declaradas con = no es necesario explicitar el tipo
+    fun esVersatil() =
+        certificaciones.size >= 3
+                && this.certificacionesDeProducto() >= 1
+                && this.otrasCertificaciones() >= 1
 
-  // Si el tipo no está declarado y la función no devuelve nada, se asume Unit (es decir, vacío)
-  fun agregarCertificacion(certificacion: Certificacion) {
-    certificaciones.add(certificacion)
-  }
+    // Si el tipo no está declarado y la función no devuelve nada, se asume Unit (es decir, vacío)
+    fun agregarCertificacion(certificacion: Certificacion) {
+        certificaciones.add(certificacion)
+    }
 
-  fun esFirme() = this.puntajeCertificaciones() >= 30
+    fun esFirme() = this.puntajeCertificaciones() >= 30
 
-  fun certificacionesDeProducto() = certificaciones.count { it.esDeProducto }
-  fun otrasCertificaciones() = certificaciones.count { !it.esDeProducto }
+    fun certificacionesDeProducto() = certificaciones.count { it.esDeProducto }
+
+    fun otrasCertificaciones() = certificaciones.count { !it.esDeProducto }
+
+    fun puntajeCertificaciones() = certificaciones.sumBy { c -> c.puntaje }
+
+    abstract fun esInfluyente(): Boolean
 
-  fun puntajeCertificaciones() = certificaciones.sumBy { c -> c.puntaje }
 }
 
 // En los parámetros, es obligatorio poner el tipo
 class VendedorFijo(val ciudadOrigen: Ciudad) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return ciudad == ciudadOrigen
-  }
+
+    override fun puedeTrabajarEn(ciudad: Ciudad): Boolean = ciudad == ciudadOrigen
+
+    override fun esInfluyente(): Boolean = false
+
 }
 
 // A este tipo de List no se le pueden agregar elementos una vez definida
 class Viajante(val provinciasHabilitadas: List<Provincia>) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return provinciasHabilitadas.contains(ciudad.provincia)
-  }
+
+    override fun puedeTrabajarEn(ciudad: Ciudad): Boolean =
+        provinciasHabilitadas.contains(ciudad.provincia)
+
+    override fun esInfluyente(): Boolean = provinciasHabilitadas.sumBy { it.poblacion } >= 10000
+
 }
 
 class ComercioCorresponsal(val ciudades: List<Ciudad>) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return ciudades.contains(ciudad)
-  }
+
+    override fun puedeTrabajarEn(ciudad: Ciudad): Boolean = ciudades.contains(ciudad)
+
+    override fun esInfluyente(): Boolean {
+        var esInfluyente = false
+        val provinciasDondeTrabaja = ciudades.map { it.provincia }.toSet()
+        val ciudadesDondeTrabaja = ciudades.toSet()
+        if (provinciasDondeTrabaja.count() >= 3 || ciudadesDondeTrabaja.count() >= 5) {
+            esInfluyente = true
+        }
+        return esInfluyente
+    }
```

Comment:
Uy, este método está un poco fulero. 

En principio, siempre tratá de resolverlo **sin** utilizar variables como flag/bandera que cambian de estado... eso puede tener sentido si estás programando en C y buscás optimizar, pero no tiene sentido en una materia como la nuestra.

Lo que te recomiendo es que dividas en subtareas más chicas el problema, creando métodos que lo resuelvan. Por ejemplo: `fun provinciasDondeTrabaja() = ciudades.map { it.provincia }.toSet()` y luego los utilices para construir tu solución. Haciendo eso, podría quedar así:

```suggestion
    override fun esInfluyente() = provinciasDondeTrabaja().count() >= 3 || ciudadesDondeTrabaja().count() >= 5
    
    fun provinciasDondeTrabaja() = ciudades.map { it.provincia }.toSet()
    fun ciudadesDondeTrabaja() = ciudades.toSet()
```